test/test_runner.go

```diff
@@ -362,14 +420,14 @@ func (r *TestRunner) interpreterContractValueHandler(
 			return contract
 
 		case stdlib.TestContractLocation:
-			testFramework := NewEmulatorBackend(
+			r.backend = NewEmulatorBackend(
```

Comment:
We could avoid this as mentioned in: https://github.com/onflow/cadence-tools/pull/114#discussion_r1205959992

Reply:
That's a great catch :100: I added it merely for testing purposes, because I need the changes from https://github.com/onflow/cadence/pull/2505, to be able it to test it properly from the Cadence level.

Reply:
I will rework the test cases, once the prerequisite PR is merged and a Cadence release includes these changes.

Reply:
By the way, what do you think about making the `Blockchain` a singleton? Regardless of that, we can also expose the `Blockchain.ReloadBlockchain‎` method, on the testing framework, as Austin has suggested in: https://github.com/orgs/onflow/discussions/1330

> Run some function after every test, or a way to snapshot and restore state
  Based on documentation, tearDown runs after all tests have been run, but I like to write my tests in such a way that each one is its own isolated sandbox. Right now, I don't have an easy way to basically "reset" the state of the chain before running every test. It's a little weird to me that I can't reset the state of the backend being used to ensure I'm operating on a clean slate